tests/Message/ApplePayAuthorizeRequestTest.php

```diff
@@ -0,0 +1,176 @@
+        $this->assertSame($this->applicationUrl, $data['initiativeContext']);
+    }
+
+    /**
+     * @return void
+     */
+    public function testValidationUrl()
+    {
+        $request = Mocker::mock('\Omnipay\Vindicia\Message\ApplePayAuthorizeRequest')->makePartial();
+        $request->initialize();
+
+        $this->assertSame($request, $request->setValidationUrl($this->validationUrl));
+        $this->assertEquals($this->validationUrl, $request->getValidationUrl());
+    }
+
+    /**
+     * @expectedException        \Omnipay\Common\Exception\InvalidRequestException
+     * @expectedExceptionMessage The validationUrl parameter is required
+     * @return                   void
+     */
+    public function testValidationUrlRequired()
+    {
+        $this->request->setValidationUrl(null);
+        $this->request->getData();
+    }
+
+    /**
+     * @return void
+     */
+    public function testMerchantIdentifier()
+    {
+        $request = Mocker::mock('\Omnipay\Vindicia\Message\ApplePayAuthorizeRequest')->makePartial();
+        $request->initialize();
+
+        $this->assertSame($request, $request->setMerchantIdentifier($this->merchantIdentifier));
+        $this->assertSame($this->merchantIdentifier, $request->getMerchantIdentifier());
+    }
+
+    /**
+     * @return void
+     */
+    public function testDisplayName()
+    {
+        $request = Mocker::mock('\Omnipay\Vindicia\Message\ApplePayAuthorizeRequest')->makePartial();
+        $request->initialize();
+
+        $this->assertSame($request, $request->setDisplayName($this->displayName));
+        $this->assertSame($this->displayName, $request->getDisplayName());
+    }
+
+    /**
+     * @return void
+     */
+    public function testApplicationUrl()
+    {
+        $request = Mocker::mock('\Omnipay\Vindicia\Message\ApplePayAuthorizeRequest')->makePartial();
+        $request->initialize();
+
+        $this->assertSame($request, $request->setApplicationUrl($this->applicationUrl));
+        $this->assertSame($this->applicationUrl, $request->getApplicationUrl());
+    }
+
+    /**
+     * @return void
+     */
+    public function testSendSuccess()
+    {
+        $this->setMockHttpResponse('ApplePayAuthorizeRequestSuccess.txt');
+        $response = $this->request->send();
+
+        $this->assertTrue($response->isSuccessful());
+        $this->assertSame('OK', $response->getMessage());
+        $this->assertSame('200', $response->getCode());
+    }
+
+    /**
+     * @return void
+     */
+    public function testSendFailure()
+    {
+        $this->setMockHttpResponse('ApplePayAuthorizeRequestFailure.txt');
+        $response = $this->request->send();
+
+        $this->assertFalse($response->isSuccessful());
+        $this->assertSame('400', $response->getCode());
+        $this->assertSame('Not Found', $response->getMessage());
+    }
+
+    /**
+     * @return void
+     */
+    public function testGetPaymentSessionObjectFailure()
```

Comment:
these two tests could just be combined with the two tests above. this one could be part of testSendFailure and testGetPaymentSessionObjectSuccess could be part of testSendSuccess. That's how most omnipay repos do it